### DISCUSSION TITLE
Netbeans as a code editor

### DISCUSSION THREAD

**Discussion by ghost** [apache/netbeans#7134] [Ideas]
### Description

Netbeans as a code editor, inspired by code editors such as Lite XL, Sublime Text,...

### Use case/motivation

A plugin to transform Netbeans into a code editor. It will make adding support for new languages much easier. Add syntax highlighting. Setup rules to build, run, and debug. Setup LSP. And done. Pretty much the same procedure as on code editors like Lite XL, Sublime Text,...

Currently, adding support for new languages is more involved. Adding syntax highlighting is easy. But setting up basic editing features of a code editor like auto-indent, braces matching, code folding, etc. still requires you to dive into Java code. Modern code editors make this very easy. And, most of the time, you need to write no code at all. You will pretty much mess with configuration files, and most of the time, you can do this via a graphical frontend.

### Related issues

No

### Are you willing to submit a pull request?

No

**Comment by Chris2011** [comment]
I'm definitely with you, I would suggest to convert this ticket into a discussion before it will be closed as not planned

**Reply by ghost** [reply]
Now it's already been converted into a discussion. What's next?

**Comment by ghost** [comment]
Something like this is already possible in Eclipse. Please have a look:

https://www.liclipse.com/

Free version: https://www.liclipse.com/text/

**Reply by Chris2011** [reply]
I like the idea what IntelliJ did. You can open any file very fast like in vs code with simple highlighting and other stuff and if you need more powerfull stuff, you can activate the powerful package to have it. The problem here, yes NetBeans doesn't have it. What comes into my mind is, that we can have the very basic NetBeans platform application with the editor which is similar if you open NetBeans without any settings and any other packages. So LSP and textmate and maybe the generic editor is the basic module what we need here. And if the user wants more, than add/activate the packages.

Luckily NetBeans has all of those functionality, the problem here is to make it exactly as we want too. I guess this is much work.

**Comment by matthiasblaesing** [comment]
@iahung2 it is not hard to find samples, it is had to find people to do the actual work, design support and most importantly maintain it.

**Reply by ghost** [reply]
Yes, I know. But an end user like me can only give you ideas.
